pkg/ext/stores/tokens/tokens.go

```diff
@@ -856,25 +858,48 @@ func (t *Store) watch(ctx context.Context, options *metav1.ListOptions) (watch.I
 					return
 				}
 
-				// skip bogus events on not-secrets
-				secret, ok := event.Object.(*corev1.Secret)
-				if !ok {
+				var token *ext.Token
+				switch event.Type {
+				case watch.Bookmark:
+					secret, ok := event.Object.(*corev1.Secret)
+					if !ok {
+						logrus.Warnf("tokens: watch: expected secret got %T", event.Object)
+						continue
+					}
+
+					token = &ext.Token{
+						ObjectMeta: metav1.ObjectMeta{
+							ResourceVersion: secret.ResourceVersion,
+						},
+					}
+				case watch.Error:
+					status, ok := event.Object.(*metav1.Status)
+					if ok {
+						logrus.Warnf("tokens: watch: received error event: %s", status.String())
+					} else {
+						logrus.Warnf("tokens: watch: received error event: %s", event.Object.GetObjectKind().GroupVersionKind().String())
+					}
 					continue
+				default: // watch.Added, watch.Modified, watch.Deleted
```

Comment:
```suggestion
				watch.Added, watch.Modified, watch.Deleted:
```

I'd explicitly check the event is  `watch.Added`, `watch.Modified` or `watch.Deleted` in the switch instead the comment. Then, we can add a default case and log an unexpected error

Reply:
will do

Reply:
done: 41a4a36eb81f18fff63445eb633e697449fcda20